### PR TITLE
Ensure final linebreak in job template editor

### PR DESCRIPTION
**Pull request by perlpunk**

### CODE REVIEW

File: assets/javascripts/job_templates.js [hunk 1]
```diff
@@ -394,7 +394,14 @@ function submitTemplateEditor(button) {
...
     var template = editor.doc.getValue();
     if (template === '') {
-        template = "products: {}\nscenarios: {}";
+        template = "products: {}\nscenarios: {}\n";
+        editor.doc.setValue(template);
+    }
+
+    // Ensure final linebreak, as files without it often need additional
+    // handling elsewhere
+    if (template.substring(template.length - 1) != "\n") {
```
Comment (Martchus) [inline assets/javascripts/job_templates.js]: I think `template.substr(-1) !== '\n'` is better.

File: assets/javascripts/job_templates.js [hunk 2]
```diff
@@ -394,7 +394,14 @@ function submitTemplateEditor(button) {
     // Reset to the minimum viable YAML if empty
     var template = editor.doc.getValue();
     if (template === '') {
-        template = "products: {}\nscenarios: {}";
+        template = "products: {}\nscenarios: {}\n";
```
Comment (kalikiana) [inline assets/javascripts/job_templates.js]: That seems redundant given the next block?

File: t/ui/13-admin.t [hunk 3]
```diff
@@ -581,7 +581,7 @@ subtest 'edit job templates' => sub() {
     wait_for_ajax;
     like($result->get_text(), qr/YAML saved!/, 'saving confirmed') or diag explain $result->get_text();
     $yaml = $driver->execute_script('return editor.doc.getValue();');
-    is($yaml, "products: {}\nscenarios: {}", 'YAML was reset to default') or diag explain $yaml;
+    is($yaml, "products: {}\nscenarios: {}\n", 'YAML was reset to default') or diag explain $yaml;
```
Comment (kalikiana) [inline t/ui/13-admin.t]: I'd like a test that asserts the added line-break on a non-default template.

File: assets/javascripts/job_templates.js (hunk 1 above)
Comment (perlpunk) [reply assets/javascripts/job_templates.js]: oh nice. I tried that with `substring` and didn't see that there is also `substr` which accepts a negative offset

Comment (perlpunk) [reply assets/javascripts/job_templates.js]: done

File: assets/javascripts/job_templates.js (hunk 2 above)
Comment (perlpunk) [reply assets/javascripts/job_templates.js]: I changed the next `if` to `else if`

File: t/ui/13-admin.t (hunk 3 above)
Comment (perlpunk) [reply t/ui/13-admin.t]: done

**Pull request merged**